### ISSUE TITLE
Powers never resolve / Cant respawn.

### ISSUE THREAD

**Issue by Mireball** [QuantumXenon/origins-plus-plus#382]
**Loader**
Forge

**Versions**
- Minecraft Version: 1.20.1
- Mod Loader Version: 47.3.1
- Origins Version: 1.10.0.9
- Origins++ Version: 2.2
- Pehkui Version: 3.8.2

**Origin**
Any/all origins that teleport you to a different dimension (Nether/End) when spawning.

**Notes**
When first attempting to spawn in, the game will get stuck on the "Resolving powers" loading screen. Reloading the game "fixes" it, however I will always spawn in the Overworld. Moreover, if I die while playing a race that spawns in a different dimension, I can no longer respawn (option is greyed out. Only way to bypass this is to turn on immediate respawn after death.)

**Comment by Mireball** [comment]
Forgot to add logs
[latest.log](https://github.com/user-attachments/files/15796187/latest.log)

**Comment by Diego13303** [comment]
Getting this issue with Broodmother. I think it happened with Sand Walker too, so it seems like most/all Origins that spawn you somewhere other than normal, wether it´d be a dimension, biome or structure, have this issue. The respawn button does ungray after waiting a while but its definetly annoying

Clicking "Title Screen" and then trying to respawn afterwards respawns you without an Origin selected

**Comment by QuantumXenon** [comment]
@Diego13303 Yes, this issue occurs when the game hangs trying to find a spawnpoint for the player. If it is set to a biome or structure, the game can take too long to find a valid location.

**Comment by QuantumXenon** [comment]
Merging #320 with (now a pinned issue)

**Comment by kevinatorrrr** [comment]
Is there any fix for it? I'm experiencing the same issue rn with origins++ ver 2.4 on forge on mc ver 1.20.1. on singleplayer it works normally but it doesn't work on server
